reana_cluster/backends/kubernetes/k8s.py

```diff
@@ -0,0 +1,510 @@
+                # templates folder
+                # TODO: Better to combine templates or populated templates in
+                # Python code for improved extensibility?
+                # Just drop a .yaml template and add necessary to config.yaml
+                # without changing anything?
+                # template = env.get_template('deployments/complete.yaml')
+                template = env.get_template('complete.yaml')
+
+                components = cluster_spec['components']
+                server_img = components['reana-server']['image']
+                jc_img = components['reana-job-controller']['image']
+                wfc_img = components['reana-workflow-controller']['image']
+                wm_img = components['reana-workflow-monitor']['image']
+                mb_img = components['reana-message-broker']['image']
+                wfe_img = components['reana-workflow-engine-yadage']['image']
+
+                # Render the template using given backend config parameters
+                cluster_conf = template.render(backend_conf_parameters,
+                                               SERVER_IMAGE=server_img,
+                                               JOB_CONTROLLER_IMAGE=jc_img,
+                                               WORKFLOW_CONTROLLER_IMAGE=wfc_img,
+                                               WORKFLOW_MONITOR_IMAGE=wm_img,
+                                               MESSAGE_BROKER_IMAGE=mb_img,
+                                               WORKFLOW_ENGINE_IMAGE=wfe_img,
+                                               )
+
+                # Strip empty lines for improved readability
+                cluster_conf = '\n'.join(
+                    [line for line in cluster_conf.splitlines() if
+                     line.strip()])
+
+                # FIXME: Print the whole configuration in a loop
+                # Now prints just memory address of generator object
+                logging.debug('Loaded K8S config successfully: \n {}'
+                              .format(yaml.load_all(cluster_conf)))
+
+        except TemplateNotFound as e:
+            logging.info(
+                'Something wrong when fetching K8S config file templates from '
+                '{filepath} : \n'
+                '{error}'.format(
+                    filepath=cls._conf['templates_folder'],
+                    error=e.strerror))
+            raise e
+
+        except IOError as e:
+            logging.info(
+                'Something wrong when reading K8S config parameters-file from '
+                '{filepath} : \n'
+                '{error}'.format(filepath=be_conf_params_fp,
+                                 error=e.strerror))
+            raise e
+
+        # Return a generator containing independent YAML files split with `---`
+        # as Python objects.
+        # TODO: Test using `safe_load_all` -method
+        return yaml.load_all(cluster_conf)
+
+    def _cluster_running(self):
```

Comment:
another naming thingy, when I was reading `init()` method and I saw that you check if the `cluster` is running I thought "how is it going to be running if you want to initialize it"

This makes me think more about using the `backend` word (as the class name is `KubernetesBackend` and so the abstraction). As we have previously talked, I am not very comfortable with the `backend` word but at least we don't confuse the infrastructure and "our" cluster.

Reply:
> "how is it going to be running if you want to initialize it"
True :)
> I am not very comfortable with the backend word but at least we don't confuse the infrastructure and "our" cluster.
Yes, I agree.

I will come up with some solution.